shell/platform/darwin/common/framework/Headers/FlutterDartProject.h

```diff
@@ -6,37 +6,67 @@
 #define FLUTTER_FLUTTERDARTPROJECT_H_
 
 #import <Foundation/Foundation.h>
+#import <TargetConditionals.h>
 
 #import "FlutterMacros.h"
 
-NS_ASSUME_NONNULL_BEGIN
-
 /**
  * A set of Flutter and Dart assets used by a `FlutterEngine` to initialize execution.
+ *
+ * TODO(stuartmorgan): Align API with FlutterDartProject, and combine.
  */
 FLUTTER_DARWIN_EXPORT
 @interface FlutterDartProject : NSObject
 
 /**
  * Initializes a Flutter Dart project from a bundle.
+ *
+ * The bundle must either contain a flutter_assets resource directory, or set the Info.plist key
+ * FLTAssetsPath to override that name (if you are doing a custom build using a different name).
+ *
+ * @param bundle The bundle containing the Flutter assets directory. If nil, the App framework
+ *               created by Flutter will be used.
  */
-- (instancetype)initWithPrecompiledDartBundle:(nullable NSBundle*)bundle NS_DESIGNATED_INITIALIZER;
+- (nonnull instancetype)initWithPrecompiledDartBundle:(nullable NSBundle*)bundle
+    NS_DESIGNATED_INITIALIZER;
 
+#if TARGET_OS_IPHONE
 /**
  * Unavailable - use `init` instead.
  */
-- (instancetype)initFromDefaultSourceForConfiguration FLUTTER_UNAVAILABLE("Use -init instead.");
+- (nonnull instancetype)
+    initFromDefaultSourceForConfiguration FLUTTER_UNAVAILABLE("Use -init instead.");
+
+/**
+ * Returns the default identifier for the bundle where we expect to find the Flutter Dart
+ * application.
+ */
++ (nonnull NSString*)defaultBundleIdentifier;
+
+#endif
+
+#if TARGET_OS_OSX
+/**
+ * An NSArray of NSStrings to be passed as command line arguments to the Dart entrypoint.
+ *
+ * If this is not explicitly set, this will default to the contents of
+ * [NSProcessInfo arguments], without the binary name.
+ *
+ * Set this to nil to pass no arguments to the Dart entrypoint.
+ */
+@property(nonatomic, nullable, copy) NSArray<NSString*>* dartEntrypointArguments;
+#endif
```

Comment:
@jmagman Seems `API_UNAVAILABLE` causes [lint error](https://ci.chromium.org/ui/p/flutter/builders/try/Mac%20iOS%20clang-tidy/7572/overview), any ideas? 
```
❌ Failures for clang-tidy on /Volumes/Work/s/w/ir/cache/builder/src/flutter/shell/platform/darwin/ios/framework/Source/FlutterDartProject.mm:
/Volumes/Work/s/w/ir/cache/builder/src/flutter/shell/platform/darwin/ios/framework/Source/FlutterDartProject.mm:241:1: error: 'FlutterDartProject' lacks a 'dealloc' instance method but must release '_dartEntrypointArguments' [clang-analyzer-osx.cocoa.Dealloc,-warnings-as-errors]
@implementation FlutterDartProject {
^
/Volumes/Work/s/w/ir/cache/builder/src/flutter/shell/platform/darwin/ios/framework/Source/FlutterDartProject.mm:241:1: note: 'FlutterDartProject' lacks a 'dealloc' instance method but must release '_dartEntrypointArguments'
Suppressed 2655 warnings (2643 in non-user code, 12 NOLINT).
Use -header-filter=.* to display errors from all non-system headers. Use -system-headers to display errors from system headers as well.
1 warning treated as error
```

Reply:
That's annoying. I was able to work around this by adding `@dynamic` to the implementation here

https://github.com/flutter/engine/blob/3e5f27d4812efa0b8cd4b1fc6d7c11f80fcfe56b/shell/platform/darwin/ios/framework/Source/FlutterDartProject.mm#L288

Something like:

```objc
// This property is marked unavailable on iOS in the common header.
// That doesn't seem to be enough to prevent this property from being synthesized.
// Mark dynamic to avoid warnings.
@dynamic dartEntrypointArguments;
```

Reply:
Done